segmentation violation:  main.getDateAndDigestAndSize

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**
$ buildah images segfailts in main.getDateAndDigestAndSize after loading minified oci-archives created with smith via `buildah pull`

```

buildah images
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x55f7b4f72a5b]

goroutine 1 [running]:
main.getDateAndDigestAndSize(0x55f7b570ab60, 0xc000034188, 0xc000097980, 0x55f7b572d500, 0xc000476480, 0xc000387600, 0x40,000018370, 0x47, 0xc00016e7d0, ...)
        github.com/containers/buildah@/cmd/buildah/common.go:181 +0x49b
main.outputImages(0x55f7b570ab60, 0xc000034188, 0xc000097980, 0x55f7b572d500, 0xc000476480, 0xc00001ba80, 0x1c, 0x1c, 0x0,, ...)
        github.com/containers/buildah@/cmd/buildah/images.go:286 +0x7cd
main.imagesCmd(0xc0003cf400, 0x55f7b6013138, 0x0, 0x0, 0xc0003b3230, 0x0, 0x0)
        github.com/containers/buildah@/cmd/buildah/images.go:178 +0x2a6
main.init.7.func1(0xc0003cf400, 0x55f7b6013138, 0x0, 0x0, 0x0, 0x0)
        github.com/containers/buildah@/cmd/buildah/images.go:97 +0x54
github.com/spf13/cobra.(*Command).execute(0xc0003cf400, 0x55f7b6013138, 0x0, 0x0, 0xc0003cf400, 0x55f7b6013138)
        github.com/spf13/cobra@v0.0.5/command.go:826 +0x462
github.com/spf13/cobra.(*Command).ExecuteC(0x55f7b5f8bbc0, 0x6, 0xc000034107, 0x8)
        github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fd
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
        github.com/containers/buildah@/cmd/buildah/main.go:142 +0xe3
ERRO exit status 2                                
```

**Steps to reproduce the issue:**
1. Create an OCI archive. I've used `podman push docker.io/library/fedora oci-archive:fedora.tar.gz:latest`
2. Minify the image using smith. My smith.yaml is:
```
type: oci
package: fedora.tar.gz
paths:
   - /usr/bin/python3
cmd:
   - /usr/bin/python3
```
3. pull the resulting image `buildah pull oci-archive:image.tar.gz:latest`
4. display all images with `buildah images`

**Describe the results you received:**
I've received a segfault.

**Describe the results you expected:**
I've expected that `buildag images` would work just fine. For reference `podman images` does.

**Output of `rpm -q buildah` or `apt list buildah`:**

```
Name            : buildah
Version         : 1.13.1-1
Description     : A tool which facilitates building OCI images
Architecture    : x86_64
URL             : https://github.com/containers/buildah
Licenses        : Apache
Groups          : None
Provides        : None
Depends On      : ostree  runc  skopeo  slirp4netns
Optional Deps   : None
Required By     : None
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 36.42 MiB
Packager        : Bartłomiej Piotrowski <bpiotrowski@archlinux.org>
Build Date      : Wed Jan 15 15:01:15 2020
Install Date    : Wed Jan 22 15:14:30 2020
Install Reason  : Explicitly installed
Install Script  : No
Validated By    : Signature
```

**Output of `buildah version`:**

```
LANG= buildah version
Version:         1.13.1
Go Version:      go1.13.6
Image Spec:      1.0.1-dev
Runtime Spec:    1.0.1-dev
CNI Spec:        0.4.0
libcni Version:  v0.7.1
image Version:   5.1.0
Git Commit:      836d6d2e
Built:           Wed Jan 15 15:01:15 2020
OS/Arch:         linux/amd64
```

**Output of `podman version` if reporting a `podman build` issue:**

```
podman version
Version:            1.7.0
RemoteAPI Version:  1
Go Version:         go1.13.5
Git Commit:         b7ce1157b00af09f4a09e39b377aa3abff46ee05-dirty
Built:              Sat Jan 11 13:49:39 2020
OS/Arch:            linux/amd64
```

**Output of `cat /etc/*release`:**
I am on arch linux

**Output of `uname -a`:**

```
Linux akela 5.4.15-arch1-1 #1 SMP PREEMPT Sun, 26 Jan 2020 09:48:50 +0000 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
# This file is is the configuration file for all tools
# that use the containers/storage library.
# See man 5 containers-storage.conf for more information
# The "container storage" table contains all of the server options.
[storage]

# Default Storage Driver
driver = "overlay"

# Temporary storage location
runroot = "/var/run/containers/storage"

# Primary Read/Write location of container storage
graphroot = "/var/lib/containers/storage"

[storage.options]
# Storage options to be passed to underlying storage drivers

# AdditionalImageStores is used to pass paths to additional Read/Only image stores
# Must be comma separated list.
additionalimagestores = [
]

# Size is used to set a maximum size of the container image.  Only supported by
# certain container storage drivers.
size = ""

# Path to an helper program to use for mounting the file system instead of mounting it
# directly.
#mount_program = "/usr/bin/fuse-overlayfs"

# OverrideKernelCheck tells the driver to ignore kernel checks based on kernel version
override_kernel_check = "true"

# mountopt specifies comma separated list of extra mount options
mountopt = "nodev"

# Remap-UIDs/GIDs is the mapping from UIDs/GIDs as they should appear inside of
# a container, to UIDs/GIDs as they should appear outside of the container, and
# the length of the range of UIDs/GIDs.  Additional mapped sets can be listed
# and will be heeded by libraries, but there are limits to the number of
# mappings which the kernel will allow when you later attempt to run a
# container.
#
# remap-uids = 0:1668442479:65536
# remap-gids = 0:1668442479:65536

# Remap-User/Group is a name which can be used to look up one or more UID/GID
# ranges in the /etc/subuid or /etc/subgid file.  Mappings are set up starting
# with an in-container ID of 0 and the a host-level ID taken from the lowest
# range that matches the specified name, and using the length of that range.
# Additional ranges are then assigned, using the ranges which specify the
# lowest host-level IDs first, to the lowest not-yet-mapped container-level ID,
# until all of the entries have been used for maps.
#
# remap-user = "storage"
# remap-group = "storage"

[storage.options.thinpool]
# Storage Options for thinpool

# autoextend_percent determines the amount by which pool needs to be
# grown. This is specified in terms of % of pool size. So a value of 20 means
# that when threshold is hit, pool will be grown by 20% of existing
# pool size.
# autoextend_percent = "20"

# autoextend_threshold determines the pool extension threshold in terms
# of percentage of pool size. For example, if threshold is 60, that means when
# pool is 60% full, threshold has been hit.
# autoextend_threshold = "80"

# basesize specifies the size to use when creating the base device, which
# limits the size of images and containers.
# basesize = "10G"

# blocksize specifies a custom blocksize to use for the thin pool.
# blocksize="64k"

# directlvm_device specifies a custom block storage device to use for the
# thin pool. Required if you setup devicemapper.
# directlvm_device = ""

# directlvm_device_force wipes device even if device already has a filesystem.
# directlvm_device_force = "True"

# fs specifies the filesystem type to use for the base device.
# fs="xfs"

# log_level sets the log level of devicemapper.
# 0: LogLevelSuppress 0 (Default)
# 2: LogLevelFatal
# 3: LogLevelErr
# 4: LogLevelWarn
# 5: LogLevelNotice
# 6: LogLevelInfo
# 7: LogLevelDebug
# log_level = "7"

# min_free_space specifies the min free space percent in a thin pool require for
# new device creation to succeed. Valid values are from 0% - 99%.
# Value 0% disables
# min_free_space = "10%"

# mkfsarg specifies extra mkfs arguments to be used when creating the base.
# device.
# mkfsarg = ""

# use_deferred_removal marks devicemapper block device for deferred removal.
# If the thinpool is in use when the driver attempts to remove it, the driver 
# tells the kernel to remove it as soon as possible. Note this does not free
# up the disk space, use deferred deletion to fully remove the thinpool.
# use_deferred_removal = "True"

# use_deferred_deletion marks thinpool device for deferred deletion.
# If the device is busy when the driver attempts to delete it, the driver
# will attempt to delete device every 30 seconds until successful.
# If the program using the driver exits, the driver will continue attempting
# to cleanup the next time the driver is used. Deferred deletion permanently
# deletes the device and all data stored in device will be lost.
# use_deferred_deletion = "True"

# xfs_nospace_max_retries specifies the maximum number of retries XFS should
# attempt to complete IO when ENOSPC (no space) error is returned by
# underlying storage device.
# xfs_nospace_max_retries = "0"

# If specified, use OSTree to deduplicate files with the overlay backend
ostree_repo = ""

# Set to skip a PRIVATE bind mount on the storage home directory.  Only supported by
# certain container storage drivers
skip_mount_home = "false"

```
~/.config/containers/storage.conf
```
[storage]
  driver = "overlay"
  runroot = "/run/user/1000/containers"
  graphroot = "/home/tobias/.local/share/containers/storage"
  [storage.options]
    size = ""
    remap-uids = ""
    remap-gids = ""
    ignore_chown_errors = ""
    remap-user = ""
    remap-group = ""
    skip_mount_home = ""
    mount_program = "/usr/bin/fuse-overlayfs"
    mountopt = ""
    [storage.options.aufs]
      mountopt = ""
    [storage.options.btrfs]
      min_space = ""
      size = ""
    [storage.options.thinpool]
      autoextend_percent = ""
      autoextend_threshold = ""
      basesize = ""
      blocksize = ""
      directlvm_device = ""
      directlvm_device_force = ""
      fs = ""
      log_level = ""
      min_free_space = ""
      mkfsarg = ""
      mountopt = ""
      size = ""
      use_deferred_deletion = ""
      use_deferred_removal = ""
      xfs_nospace_max_retries = ""
    [storage.options.overlay]
      ignore_chown_errors = ""
      mountopt = ""
      mount_program = "/usr/bin/fuse-overlayfs"
      size = ""
      skip_mount_home = ""
    [storage.options.vfs]
      ignore_chown_errors = ""
    [storage.options.zfs]
      mountopt = ""
      fsname = ""
      size = ""
```

--- a/cmd/buildah/common.go
+++ b/cmd/buildah/common.go
@@ -193,7 +193,7 @@ func getDateAndDigestAndSize(ctx context.Context, sys *types.SystemContext, stor
 	inspectable, inspectableErr := image.FromUnparsedImage(ctx, sys, image.UnparsedInstance(img, nil))
 	if inspectableErr == nil && inspectable != nil {
 		inspectInfo, inspectErr := inspectable.Inspect(ctx)
-		if inspectErr == nil && inspectInfo != nil {
+		if inspectErr == nil && inspectInfo != nil && inspectInfo.Created != nil {
 			created = *inspectInfo.Created
 		}
 	}

--- a/tests/images.bats
+++ b/tests/images.bats
@@ -170,3 +170,93 @@ load helpers
   run_buildah images --digests
   expect_output --substring "sha256:"
 }
+
+@test "images in OCI format with no creation dates" {
+  mkdir -p $TESTDIR/blobs/sha256
+
+  # Create a layer.
+  dd if=/dev/zero bs=512 count=2 of=$TESTDIR/blob
+  layerdigest=$(sha256sum $TESTDIR/blob | awk '{print $1}')
+  layersize=$(stat -c %s $TESTDIR/blob)
+  mv $TESTDIR/blob $TESTDIR/blobs/sha256/${layerdigest}
+
+  # Create a configuration blob that doesn't include a "created" date.
+  now=$(TZ=UTC date +%Y-%m-%dT%H:%M:%S.%NZ)
+  arch=$(go env GOARCH)
+  cat > $TESTDIR/blob << EOF
+  {
+    "architecture": "$arch",
+    "os": "linux",
+    "config": {
+        "Env": [
+            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+        ],
+        "Cmd": [
+            "sh"
+        ]
+    },
+    "rootfs": {
+        "type": "layers",
+        "diff_ids": [
+            "sha256:${layerdigest}"
+        ]
+    },
+    "history": [
+        {
+            "created": "${now}",
+            "created_by": "/bin/sh -c #(nop) ADD file:${layerdigest} in / "
+        }
+    ]
+  }
+EOF
+  configdigest=$(sha256sum $TESTDIR/blob | awk '{print $1}')
+  configsize=$(stat -c %s $TESTDIR/blob)
+  mv $TESTDIR/blob $TESTDIR/blobs/sha256/${configdigest}
+
+  # Create a manifest for that configuration blob and layer.
+  cat > $TESTDIR/blob << EOF
+  {
+    "schemaVersion": 2,
+    "config": {
+        "mediaType": "application/vnd.oci.image.config.v1+json",
+        "digest": "sha256:${configdigest}",
+        "size": ${configsize}
+    },
+    "layers": [
+        {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar",
+            "digest": "sha256:${layerdigest}",
+            "size": ${layersize}
+        }
+    ]
+  }
+EOF
+  manifestdigest=$(sha256sum $TESTDIR/blob | awk '{print $1}')
+  manifestsize=$(stat -c %s $TESTDIR/blob)
+  mv $TESTDIR/blob $TESTDIR/blobs/sha256/${manifestdigest}
+
+  # Add the manifest to the image index.
+  cat > $TESTDIR/index.json << EOF
+  {
+    "schemaVersion": 2,
+    "manifests": [
+        {
+            "mediaType": "application/vnd.oci.image.manifest.v1+json",
+            "digest": "sha256:${manifestdigest}",
+            "size": ${manifestsize}
+        }
+    ]
+  }
+EOF
+
+  # Mark the directory as a layout directory.
+  echo -n '{"imageLayoutVersion": "1.0.0"}' > $TESTDIR/oci-layout
+
+  # Import the image.
+  run_buildah pull oci:$TESTDIR
+
+  # Inspect the image.  We shouldn't crash.
+  run_buildah inspect ${configdigest}
+  # List images.  We shouldn't crash.
+  run_buildah images
+}

